src/main/java/com/palantir/semver/SemverVersion.java

```diff
@@ -0,0 +1,52 @@
+package com.palantir.semver;
+
+public class SemverVersion {
+
+    private final String fullVersion;
+    private final String tagVersion;
+    private final String headCommitHash;
+    private final Integer count;
+    private final Integer buildNumber;
+    private final boolean isDirty;
+    private final boolean isVersionStableRelease;
+    
+    public SemverVersion (String fullVersion, String tagVersion, String headCommitHash,
+            Integer count, Integer buildNumber, boolean isDirty, boolean isVersionStableRelease) {
+        this.fullVersion = fullVersion;
+        this.tagVersion = tagVersion;
+        this.headCommitHash = headCommitHash;
+        this.count = count;
+        this.buildNumber = buildNumber;
+        this.isDirty = isDirty;
+        this.isVersionStableRelease = isVersionStableRelease;
+    }
+    
+    public String toString() {
+        return this.fullVersion;
```

Comment:
Yes, simply because of the complicated nature of the plugin, I'm not gonna go in there and fix it to make it less sad.  There are 4 or 5 levels of method calls just to get to that logic.

If we ever fix this plugin, I agree, the object should build it.